src/stores/right-panel/RightPanelStore.ts

```diff
@@ -0,0 +1,334 @@
+        const refireParams = redirect?.panelState ?? panelState;
+
+        // Checks for wrong SetRightPanelPhase requests
+        if (!this.isPhaseActionIsValid(targetPhase)) return;
+
+        let roomCache = this.byRoom[rId];
+        if (!!roomCache) {
+            // append new phase
+            roomCache.history.push({ state: refireParams, phase: targetPhase });
+            roomCache.isOpen = allowClose ? roomCache.isOpen : true;
+        } else {
+            // create new phase
+            roomCache = {
+                history: [{ phase: targetPhase, state: refireParams }],
+                // if there was no right panel store object the the panel was closed -> keep it closed, except if allowClose==false
+                isOpen: !allowClose,
+            };
+        }
+
+        this.emitAndUpdateSettings();
+    }
+
+    // pop right panel: removes last eelemnt from history
+    public popRightPanel(roomId: string = null) {
+        const rId = roomId ?? this.viewedRoomId;
+        const roomCache = this.byRoom[rId];
+        roomCache.history.pop();
+        this.emitAndUpdateSettings();
+    }
+
+    public togglePanel(roomId: string = null) {
+        const rId = roomId ?? this.viewedRoomId;
+        this.byRoom[rId].isOpen = !this.byRoom[rId].isOpen;
+        this.emitAndUpdateSettings();
+    }
+
+    // Private
+
+    private loadCacheFromSettings() {
+        const room = this.mxClient?.getRoom(this.viewedRoomId);
+        if (!!room) {
+            this.global = this.global ?? convertToStateRoom(SettingsStore.getValue("RightPanel.phasesGlobal"), room);
+            this.byRoom[this.viewedRoomId] = this.byRoom[this.viewedRoomId]
+            ?? convertToStateRoom(SettingsStore.getValue("RightPanel.phases", this.viewedRoomId), room);
+        }
+    }
+    private emitAndUpdateSettings() {
+        const storePanelGlobal = convertToStoreRoom(this.global);
+        const storePanelThisRoom = convertToStoreRoom(this.byRoom[this.viewedRoomId]);
+        SettingsStore.setValue("RightPanel.phasesGlobal", null, SettingLevel.DEVICE, storePanelGlobal);
+        if (!!this.viewedRoomId) {
+            SettingsStore.setValue("RightPanel.phases",
+                this.viewedRoomId,
+                SettingLevel.ROOM_DEVICE,
+                storePanelThisRoom,
+            );
+        }
+        this.emit(UPDATE_EVENT, null);
+    }
+
+    // set rigth panel: overrides the history
+    private setRightPanelCache(targetPhase: RightPanelPhases, panelState: IPanelState = null) {
+        this.byRoom[this.viewedRoomId] = {
+            history: [{ phase: targetPhase, state: panelState ?? {} }],
+            isOpen: true,
+        };
+        this.emitAndUpdateSettings();
+    }
+
+    // CHECKS:
+    //  - RoomMemberInfo -> needs to be changed to EncryptionPanel if pendingVerificationRequestForUser
+    private getVerificationRedirect(targetPhase: RightPanelPhases, panelState):
+        { targetPhase: RightPanelPhases, panelState } {
```

Comment:
Perhaps define an interface in this module for the return type, so you can move it up a line...? This style looks strange with the function definition at the same indent level as the body.